### PR TITLE
test: assert files exist for all platforms in e2e

### PR DESCRIPTION
**Pull request by rubenruizdegauna**
Added assertions for all distributed platforms.

### CODE REVIEW

**Review by varas — APPROVED** [review]
looks great thanks!

File: test/publish-agent.sh [hunk 1]
```diff
@@ -104,7 +104,7 @@ for _ASSERT_FILE_ABSOLUTE_PATH in "${_ASSERT_FILES[@]}";do
   printf "\n- Verifying TAG %s was published into ${_ASSERT_DIR}...\n" "$TAG"
 
   aws s3 ls "${_ASSERT_FILE_ABSOLUTE_PATH}" \
-  || (printf '\nError: missing published asset: %s!\n' "${_ASSERT_FILE_ABSOLUTE_PATH}" && exit 1)
+  || printf '\nError: missing published asset: %s!\n' "${_ASSERT_FILE_ABSOLUTE_PATH}"
```
Comment (brushknight) [inline test/publish-agent.sh:107]: how will we exit if have a failure? we should indicate it on the job run 🤔

Comment (rubenruizdegauna) [reply test/publish-agent.sh:107]: true :(  will store in var and check later, good catch!

Comment (varas) [reply test/publish-agent.sh:107]: ops! that's right

**Pull request merged**